source/extensions/filters/network/thrift_proxy/binary_protocol_impl.cc

```diff
@@ -52,13 +51,13 @@ bool BinaryProtocolImpl::readMessageBegin(Buffer::Instance& buffer, std::string&
   buffer.drain(8);
 
   if (name_len > 0) {
-    name.assign(std::string(static_cast<char*>(buffer.linearize(name_len)), name_len));
+    metadata.setMethodName(std::string(static_cast<char*>(buffer.linearize(name_len)), name_len));
```

Comment:
[minor] The cast ideally should be to `const char*` rather than `char*`.

Reply:
should there be a check for `name_len`'s max value? Otherwise a ill formed message could cause Envoy to allocated up to 4gb...